web/modules/custom/lit_search/src/Controller/SearchController.php

```diff
@@ -113,27 +121,31 @@ private function transformResults(array $groups): array {
     return $result;
   }
 
-
   /**
-   *
+   * Find entities from group items.
    *
    * @param array $hits
+   *   A list of hits.
    *
    * @return array
+   *   A list of matched rendered entities.
    */
   private function findEntitiesFromGroupItems(array $hits): array {
     $result = [];
 
     foreach ($hits as $hit) {
       // Get entity type and id form the elastic search id.
-      preg_match('/(\w*):(\w*)\/(\d*):(\w*)/', $hit['_id'], $matches);
-
-      try {
-        $entity = \Drupal::entityTypeManager()->getStorage($matches[2])->load($matches[3]);
-        $result[] = $this->renderEntity($entity, self::VIEW_MODE);;
-      }
-      catch (\Exception $e) {
-        // No action a item simply was not loaded.
+      if ($hit['_id']) {
```

Comment:
Could this be more explicit by using array_key_exists() or isset()

Reply:
Changed